examples/gno.land/r/demo/profile/profile.gno

```diff
@@ -0,0 +1,312 @@
+}
+
+// Display country from address or username
+func DisplayCountry(address std.Address, username string) string {
+	if address != "" {
+		if value, exists := profiles.Get(address.String()); exists {
+			return value.(UserProfile).Country
+		}
+	} else if username != "" {
+		if addr, exists := usernameIndex.Get(username); exists {
+			if value, exists := profiles.Get(addr.(string)); exists {
+				return value.(UserProfile).Country
+			}
+		}
+	}
+	return "Address or Username not found"
+}
+
+// Display city from address or username
+func DisplayCity(address std.Address, username string) string {
+	if address != "" {
+		if value, exists := profiles.Get(address.String()); exists {
+			return value.(UserProfile).City
+		}
+	} else if username != "" {
+		if addr, exists := usernameIndex.Get(username); exists {
+			if value, exists := profiles.Get(addr.(string)); exists {
+				return value.(UserProfile).City
+			}
+		}
+	}
+	return "Address or Username not found"
+}
+
+// Display modification history specific to a field from address or username
+func DisplayModificationHistory(address std.Address, username, field string) string {
+	if address != "" {
+		if value, exists := profiles.Get(address.String()); exists {
+			if value.(UserProfile).ModificationHistory != nil {
+				if changes, exists := value.(UserProfile).ModificationHistory[field]; exists {
+					var b bytes.Buffer
+					for _, change := range changes {
+						b.WriteString(ufmt.Sprintf("%s\n", change))
+					}
+					return b.String()
+				}
+			}
+		}
+	} else if username != "" {
+		if addr, exists := usernameIndex.Get(username); exists {
+			if value, exists := profiles.Get(addr.(string)); exists {
+				if value.(UserProfile).ModificationHistory != nil {
+					if changes, exists := value.(UserProfile).ModificationHistory[field]; exists {
+						var b bytes.Buffer
+						for _, change := range changes {
+							b.WriteString(ufmt.Sprintf("%s\n", change))
+						}
+						return b.String()
+					}
+				}
+			}
+		}
+	}
+	return "Address or Username not found or no modifications recorded for the field"
+}
+
+func Render(path string) string {
+	if path == "" {
+		return renderHome()
+	}
+
+	return "unknown path"
+}
+
+func renderHome() string {
+	var b bytes.Buffer
+
+	profiles.Iterate("", "", func(key string, value interface{}) bool {
+		profile := value.(UserProfile)
+		b.WriteString(ufmt.Sprintf("Username: %s\n\n", profile.Username))
+		b.WriteString(ufmt.Sprintf("Address: %s\n\n", profile.Address))
+		b.WriteString(ufmt.Sprintf("AvatarURL: %s\n\n", profile.AvatarURL))
+		b.WriteString(ufmt.Sprintf("Age: %s\n\n", profile.Age))
+		b.WriteString(ufmt.Sprintf("Gender: %s\n\n", profile.Gender))
+		b.WriteString(ufmt.Sprintf("Website: %s\n\n", profile.Website))
+		b.WriteString(ufmt.Sprintf("Country: %s\n\n", profile.Country))
+		b.WriteString(ufmt.Sprintf("City: %s\n\n", profile.City))
+
+		// Properly iterate over the modification history map
+		if profile.ModificationHistory != nil {
```

Comment:
Instead of listing all modification changes for every profile, which will quickly make the VM run out of gas when there is a high number of users, maybe its better to make a the username clickable, and make it take the user to separate page for displaying more info on this. This is exactly what the path argument is used for in Render, and you can take a look at an example of generating these kinds of links in `p/demo/blog` & `r/demo/userbook`.

Reply:
The [commit](https://github.com/gnolang/gno/pull/1983/commits/a0a10279605059f6e8af4abb5ea3b3ef212bf8ec) to modify the display using `mux` and the [commit](https://github.com/gnolang/gno/pull/1983/commits/9b87d5830c0548febb274c8e211e8693bf0d219e) to add a menu on the profile and modification history pages